### PR TITLE
test: e2e request queue manually switch network

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

Adds e2e spec for testing dapp provider proxy in various scenarios when the feature flag is on and off

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25271?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2627

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/request-queuing/chain-id-check.spec.js
Comment (jiexi) [inline test/e2e/tests/request-queuing/chain-id-check.spec.js:1]: this file is covered by `test/e2e/tests/request-queuing/chainid-check.spec.js`

**Comment by jiexi** [comment]
@metamaskbot update-policies

File: lavamoat/browserify/desktop/policy.json [hunk 1]
```diff
@@ -0,0 +1,5605 @@
+{
```
Comment (adonesky1) [inline lavamoat/browserify/desktop/policy.json:1]: I believe this file has been deleted on develop

**Review by shanejonas — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
LGTM. Just one question about a dangling todo?

**Pull request merged**